### PR TITLE
[Android] Fix crash due to lock on destroyed mutex

### PR DESCRIPTION
**Pull request by enen92**
## Description
Backport of https://github.com/xbmc/xbmc/pull/19232

### CODE REVIEW

File: xbmc/platform/android/activity/XBMCApp.cpp [hunk 1]
```diff
@@ -1250,6 +1250,18 @@ int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, C
...
+  // delete from m_activityResultEvents map before deleting the event
+  CSingleLock lock(m_activityResultMutex);
+  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  {
+    if ((*it)->GetRequestCode() == requestCode)
+    {
+      m_activityResultEvents.erase(it);
+      break;
+    }
+  }
```
Comment (howie-f) [inline xbmc/platform/android/activity/XBMCApp.cpp:1263]: know the v20 pr looks the same and there is no technical difference, but shouldn't we do it like this usually?
```c
for (const auto& activityResultEvent : m_activityResultEvents)
{
  if (activityResultEvent->GetRequestCode() == requestCode)
  {
    m_activityResultEvents.erase(activityResultEvent);
    break;
  }
}
```

Comment (arnova) [reply xbmc/platform/android/activity/XBMCApp.cpp:1263]: Yes C+11 iterators please

Comment (ksooo) [reply xbmc/platform/android/activity/XBMCApp.cpp:1263]: +1

Comment (enen92) [reply xbmc/platform/android/activity/XBMCApp.cpp:1263]: Thanks will do

**Comment by enen92** [comment]
Closing as the change needs to go to v20 before the backport

**Review by ksooo — APPROVED** [review]

**Pull request merged**